src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -377,12 +378,26 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                         if (logDebugBuilder.Length > 0)
                         {
-                            _logger?.LogDebug(logDebugBuilder.ToString().Trim());
+                            if (_logger != null)
+                            {
+                                _logger?.LogDebug(logDebugBuilder.ToString().Trim());
+                            } 
+                            else
+                            {
+                                AzureAppConfigurationProviderEventSource.Log.LogDebug("\n\t" + logDebugBuilder.ToString().Trim());
+                            }
                         }

```

Comment:
I hope we can put most of these logic in a helper function. This code doesn't need to know which logger it's using.